src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -7418,4 +7430,43 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="WRN_ParamsArrayInLambdaOnly_Title" xml:space="preserve">
     <value>Parameter has params modifier in lambda but not in target delegate type.</value>
   </data>
+  <data name="ERR_BadExtensionUnderlyingType" xml:space="preserve">
+    <value>The extended type may not be dynamic, a pointer, a nullable reference type, a ref struct or an extension.</value>
```

Comment:
We will need to make sure we test these (particularly pointers) in metadata. Today, you can manually create an extension method on a pointer, and we'll honor it.

Reply:
Definitely, added a note to test plan to remember to cover that.